### PR TITLE
fix: multiple renderers clean up isue

### PR DESCRIPTION
**Pull request by tejal29**
shd merge after #7698 

In this PR, 
Instead of having a `hasDeployed` to indicate if a deployed happened, code now uses a `deployManifests` which stores the last deployed manifests. 

Cleanup will use this in `dev` cycle to clean up all the manifests deployed by last iteration. 

Fixes Multiple Render clean up issue 
```
➜  multiple-renderers git:(fix_sig_and_dev_cleanup) ✗ ../../../out/skaffold dev -d gcr.io/tejal-gke1
Listing files to watch...
 - go-guestbook-backend
 - go-guestbook-frontend
Generating tags...
 - go-guestbook-backend -> gcr.io/tejal-gke1/go-guestbook-backend:v1.37.2-127-g465de3403-dirty
 - go-guestbook-frontend -> gcr.io/tejal-gke1/go-guestbook-frontend:v1.37.2-127-g465de3403-dirty
Checking cache...
 - go-guestbook-backend: Found. Tagging
 - go-guestbook-frontend: Found. Tagging
Tags used in deployment:
 - go-guestbook-backend -> gcr.io/tejal-gke1/go-guestbook-backend:v1.37.2-127-g465de3403-dirty@sha256:1aa0519da1663092bc55b4b64a7e673af164c4b9812c189959b58534e26e1ef7
 - go-guestbook-frontend -> gcr.io/tejal-gke1/go-guestbook-frontend:v1.37.2-127-g465de3403-dirty@sha256:c1a4b85cb88f8f4da283911c04fa854c86b92775731e00e5a47b5678c8b9aa48
Starting deploy...
 - deployment.apps/frontend configured
 - service/frontend configured
 - service/go-guestbook-mongodb configured
 - service/backend configured
 - deployment.apps/backend configured
 - deployment.apps/go-guestbook-mongodb configured
Waiting for deployments to stabilize...
 - deployment/go-guestbook-mongodb is ready. [2/3 deployment(s) still pending]
 - deployment/frontend is ready. [1/3 deployment(s) still pending]
 - deployment/backend is ready.
Deployments stabilized in 6.938 seconds
Press Ctrl+C to exit
Watching for changes...
[backend] 2022/07/28 02:33:48 backend server listening on port 8081
[frontend] 2022/07/28 02:33:47 frontend server listening on port 27017
[backend] 2022/07/28 02:34:00 backend server listening on port 8081
^CCleaning up...
 - deployment.apps "frontend" deleted
 - service "frontend" deleted
 - service "go-guestbook-mongodb" deleted
 - service "backend" deleted
 - deployment.apps "backend" deleted
 - deployment.apps "go-guestbook-mongodb" deleted
```

### CODE REVIEW

File: pkg/skaffold/render/renderer/renderer.go [hunk 1]
```diff
@@ -39,18 +39,32 @@ type Renderer interface {
...
 	}
-	if renderCfg.Validate == nil && renderCfg.Transform == nil && renderCfg.Helm != nil {
-		log.Entry(context.TODO()).Debug("setting up helm renderer")
-		return helm.New(cfg, renderCfg, labels, configName)
+	if renderCfg.Validate != nil && renderCfg.Transform != nil {
+		r, err := kpt.New(cfg, renderCfg, hydrationDir, labels, configName)
+		if err != nil {
+			return nil, err
+		}
+		log.Entry(context.TODO()).Infof("setting up kpt renderer")
```
Comment (gsquared94) [inline pkg/skaffold/render/renderer/renderer.go:51]: nit: add `context.Context` function parameter and use here.

File: pkg/skaffold/render/renderer/renderer.go [hunk 2]
```diff
@@ -39,18 +39,32 @@ type Renderer interface {
 }
 
 // New creates a new Renderer object from the latestV2 API schema.
-func New(cfg render.Config, renderCfg latest.RenderConfig, hydrationDir string, labels map[string]string, usingLegacyHelmDeploy bool, command string, configName string) (Renderer, error) {
+func New(cfg render.Config, renderCfg latest.RenderConfig, hydrationDir string, labels map[string]string, usingLegacyHelmDeploy bool, command string, configName string) ([]Renderer, error) {
```
Comment (gsquared94) [inline pkg/skaffold/render/renderer/renderer.go:42]: how about creating a `RendererOptions` struct, to keep function signature small?
```go
type RendererOptions struct {
    configName string
    hydrationDir string
    labels map[string]string
    usingLegacyHelmDeploy bool
}
```

File: pkg/skaffold/render/renderer/renderer.go [hunk 3]
```diff
@@ -39,18 +39,32 @@ type Renderer interface {
...
 	if usingLegacyHelmDeploy && command != "render" {
-		return noop.New(renderCfg, cfg.GetWorkingDir(), hydrationDir, labels)
+		return []Renderer{noop.New(renderCfg, cfg.GetWorkingDir(), hydrationDir, labels)}, nil
```
Comment (gsquared94) [inline pkg/skaffold/render/renderer/renderer.go:44]: nit: should we explicitly define something like:
```
// GroupRenderer maintains the slice of all `Renderer`s defined in a single Skaffold config.
type GroupRenderer []Renderer
```

**Review by gsquared94 — APPROVED** [review]
lgtm, other than some nits.

File: pkg/skaffold/render/renderer/renderer.go (hunk 2 above)
Comment (tejal29) [reply pkg/skaffold/render/renderer/renderer.go:42]: Created a fix it issue to clean this up. 
https://github.com/GoogleContainerTools/skaffold/issues/7704

Thanks
Tejal

**Pull request merged**